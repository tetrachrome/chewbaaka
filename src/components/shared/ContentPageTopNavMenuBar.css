/**
 * ContentPageTopNavMenuBar.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Aug 14, 2020
 */

div.ContentPageTopNavMenuBarContainer {
  /* width: 100%; */
  padding: 10px 0;
  display: flex;
  justify-content: space-between;
  background-color: pink;
}

div.ContentPageTopNavMenuBarMenuListSidePseudoContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

nav ul.ContentPageTopNavMenuBarMenuList {
  list-style: none;
  display: flex;
  justify-content: flex-end;
  padding: 0;
}

nav ul.ContentPageTopNavMenuBarMenuList li {
  padding: 0.6em 0.6em;
  margin-left: 1.2em;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  text-transform: uppercase;
  font-weight: 700;
}

nav ul.ContentPageTopNavMenuBarMenuList li.ActiveItem {
  background-color: rgba(0,0,0,.05);
  color: rgba(0,0,0,.95);
  border-radius: .28571429rem;
}

nav ul.ContentPageTopNavMenuBarMenuList li.ActiveItem a {
  color: rgba(0,0,0,.95);
}

nav ul.ContentPageTopNavMenuBarMenuList li a {
  color: gray;
}

nav ul.ContentPageTopNavMenuBarMenuList li a:hover {
  color: rgba(0,0,0,.95);
}

@media (max-width: 917px) {
  div.ContentPageTopNavMenuBarContainer {
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
  }

  nav ul.ContentPageTopNavMenuBarMenuList {
    justify-content: space-between;
  }

  nav ul.ContentPageTopNavMenuBarMenuList li {
    margin: auto 0.6em;
  }
}

@media (max-width: 639px) {
  div.ContentPageTopNavMenuBarContainer {
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
  }

  div.ContentPageTopNavMenuBarLogoPseudoContainer {
    margin: 10px auto;
  }

  nav ul.ContentPageTopNavMenuBarMenuList {
    flex-direction: column;
    align-items: center;
  }

  nav ul.ContentPageTopNavMenuBarMenuList li {
    margin: 0 auto;
  }
}
