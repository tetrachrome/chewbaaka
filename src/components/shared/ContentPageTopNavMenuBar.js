/**
 * ContentPageTopNavMenuBar.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 29, 2020
 */

import React from 'react';

import { Menu } from 'semantic-ui-react'

import 'semantic-ui-css/semantic.min.css'

import './ContentPageSharedStyles.css'
import './ContentPageTopNavMenuBar.css'

export default class ContentPageTopNavMenuBar extends React.Component {

  render() {
    const menuItems = [];
    const menuItemNames = ["Home", "History", "Biology", "Ecology", "Future"]

    for (const [idx, val] of menuItemNames.entries()) {
      menuItems.push(
        <a href={"/" + val.toLowerCase()} key={idx}>
          <Menu.Item key={idx} name={val} />
        </a>
      );
    }

    return (
      <div className="ContentPageTopNavMenuBarContainer">
        <Menu secondary>
          {menuItems}
        </Menu>
      </div>
    )
  }
}
