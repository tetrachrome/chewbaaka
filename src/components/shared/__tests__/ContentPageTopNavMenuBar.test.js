/**
 * ContentPageTopNavMenuBar.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Aug 06, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageTopNavMenuBar from '../ContentPageTopNavMenuBar'

import RouterWrapped from '../../../testing/RouterWrapped'

test('renders ContentPageTopNavMenuBar component', () => {
  render(
    RouterWrapped(
      <ContentPageTopNavMenuBar />
    )
  );

  const expectedMenuItemNames = ["Home", "History", "Biology", "Ecology", "Future"];

  for (const [_, expectedMenuItemName] of expectedMenuItemNames.entries()) {
    const menuItemElement = screen.getByText(expectedMenuItemName);
    expect(menuItemElement).toBeInTheDocument();
  }
});

test('ContentPageTopNavMenuBar component snapshot', () => {
  const tree = renderer
    .create(
      RouterWrapped(
        <ContentPageTopNavMenuBar />
      )
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
