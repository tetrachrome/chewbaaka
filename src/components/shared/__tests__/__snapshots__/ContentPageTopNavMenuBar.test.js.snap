// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageTopNavMenuBar component snapshot 1`] = `
<div
  className="ContentPageTopNavMenuBarContainer"
>
  <div
    className="ui secondary menu"
  >
    <a
      href="/home"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        Home
      </div>
    </a>
    <a
      href="/history"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        History
      </div>
    </a>
    <a
      href="/biology"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        Biology
      </div>
    </a>
    <a
      href="/ecology"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        Ecology
      </div>
    </a>
    <a
      href="/future"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        Future
      </div>
    </a>
  </div>
</div>
`;
